### PR TITLE
[web] Added Storybook

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

Alternate to #3962 

Using storybook to do visual regression testing on web (in CI) per @derekstavis recommendation.

# How

Implemented storybook and CI integration.

# Test Plan

~~CI should take screenshots and compare them against changes.~~

CI should build the project without failing. Will move Loki tests to another PR.

Screen Shot 2019-04-16 at 2 46 56 AM
Screen Shot 2019-04-16 at 2 47 13 AM

### CODE REVIEW

File: apps/test-suite/.storybook/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+// you can use this file to add your custom webpack plugins, loaders and anything you like.
+// This is just the basic way to add additional webpack configurations.
+// For more information refer the docs: https://storybook.js.org/configurations/custom-webpack-config
+
+// IMPORTANT
```
Comment (ide) [inline apps/test-suite/.storybook/webpack.config.js]: Remove these comments?

File: apps/test-suite/stories/1-components/BlurView/BlurViewScreen.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  Code,
+  Description,
+  DocItem,
+  Section,
+  storiesOf,
+} from '../../ui-explorer';
+
+const image = {
+  uri:
+    'https://www.tesla.com/tesla_theme/assets/img/_vehicle_redesign/roadster_and_semi/roadster/hero-portrait@2.jpg',
```
Comment (ide) [inline apps/test-suite/stories/1-components/BlurView/BlurViewScreen.js]: download this image to the repo so we don't depend on tesla.com

File: apps/test-suite/stories/1-components/BlurView/BlurViewScreen.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+          A React component that renders a native blur view on iOS and falls back to a
+          semi-transparent view on Android. A common usage of this is for navigation bars, tab bars,
+          and modals.
+        </AppText>
+      </Description>
+      <DocItem
+        name="Importing the module"
+        example={{
+          code: `// in managed apps:
+import { BlurView } from 'expo';
```
Comment (ide) [inline apps/test-suite/stories/1-components/BlurView/BlurViewScreen.js]: We are moving to `...from 'expo-blur'` everywhere, just use the unimodule import

File: apps/test-suite/stories/2-apis/Constants/ConstantsScreen.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
+/**
+ * @flow
```
Comment (ide) [inline apps/test-suite/stories/2-apis/Constants/ConstantsScreen.js]: Remove

File: apps/test-suite/stories/ui-explorer/AppText.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
+/* eslint-disable react/prop-types */
```
Comment (ide) [inline apps/test-suite/stories/ui-explorer/AppText.js]: Remove?

File: apps/test-suite/stories/ui-explorer/insertBetween.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+const insertBetween = (item, array) =>
+  array.reduce((acc, curr, i, { length }) => {
```
Comment (ide) [inline apps/test-suite/stories/ui-explorer/insertBetween.js]: reduce the memory copying and over-abstraction

```js
const result = [];
for (const i = 0; i < array.length; i++) {
  result.push(array[i]);
  if (i !== array.length) {
    result.push(item());
  }
}
return result;
```

**Review by ide — CHANGES_REQUESTED** [review]
Wondering if this is better outside of test-suite if it's not related aside from the fact that it provides tests (also there are the visual tests from https://github.com/expo/expo/pull/3962).

File: apps/test-suite/stories/1-components/BlurView/BlurViewScreen.js (hunk 2 above)
Comment (EvanBacon) [reply apps/test-suite/stories/1-components/BlurView/BlurViewScreen.js]: oh yeah some bundler issue

File: apps/test-suite/stories/ui-explorer/insertBetween.js (hunk 6 above)
Comment (EvanBacon) [reply apps/test-suite/stories/ui-explorer/insertBetween.js]: The `ui-explorer/` part is generic, perhaps someone has rolled it into a npm package.

File: apps/storybook/package.json [hunk 7]
```diff
@@ -0,0 +1,58 @@
+{
+  "name": "storybook",
+  "version": "1.0.0",
+  "description": "Storybook for the Expo SDK",
+  "main": "__generated__/AppEntry.js",
+  "scripts": {
+    "postinstall": "expo-yarn-workspaces postinstall",
+    "storybook": "start-storybook --host 10.0.0.211 --port 50713",
```
Comment (ide) [inline apps/storybook/package.json]:
```suggestion
    "start": "start-storybook --host 10.0.0.211 --port 50713",
```
?

File: apps/storybook/.storybook/webpack.config.expo.js [hunk 8]
```diff
@@ -0,0 +1,97 @@
+const webpack = require('webpack');
```
Comment (ide) [inline apps/storybook/.storybook/webpack.config.expo.js]: Would the default Expo Web webpack config work here?

**Review by ide — COMMENTED** [review]
Do the config files need to live under .storybook? Thinking it will be hard to keep them up to date if we don't use the defaults that are managed by Expo CLI or expo-module-scripts.

File: apps/storybook/.storybook/webpack.config.expo.js (hunk 8 above)
Comment (EvanBacon) [reply apps/storybook/.storybook/webpack.config.expo.js]: no, but once I merge all of the current expo-cli PRs I'll add compatibility.

File: apps/storybook/package.json (hunk 7 above)
Comment (EvanBacon) [reply apps/storybook/package.json]: This is added when bootstrap storybook

**Pull request merged**